File: cpp_topic_tutorial/src/laser_sub.cpp
/*
 * basic topic subscriber example
 * 
 * referenced from wiki.ros.org : http://wiki.ros.org/ROS/Tutorials/WritingPublisherSubscriber%28c%2B%2B%29
 */


#include <ros/ros.h>
#include <sensor_msgs/LaserScan.h>

void sub_callback(const sensor_msgs::LaserScan &data ){
    std::cout << "Front Lidar Value : " << data.ranges[360] << std::endl;
}

int main(int argc, char** argv) {

    ros::init(argc, argv, "laser_sub_node");
    ros::NodeHandle nh;    
    ros::Subscriber sub = nh.subscribe("/scan", 1, sub_callback);

    ros::spin();

    return 0;
}

File: topic_tutorial/src/laser_sub.py
#! /usr/bin/env python

"""
2D Lidar Subscriber example

referenced from wiki.ros.org

url: http://wiki.ros.org/ROS/Tutorials/WritingPublisherSubscriber%28python%29
"""

import rospy
from sensor_msgs.msg import LaserScan


def callback(data):
    # print(data.ranges[360])
    print(data)


rospy.loginfo(
    "==== Laser Scan Subscriber node Started, move forward during 10 seconds ====\n"
)

rospy.init_node("laser_scan")
sub = rospy.Subscriber("/scan", LaserScan, callback)
rospy.spin()
